### DISCUSSION TITLE
Why does the findOneAndUpdate function does not work with comments?

### DISCUSSION THREAD

**Discussion by arthur-guesty** [Automattic/mongoose#11790] [General]
I have tried to add comment to my findAndUpdate query in different methods yet I didn't manage to do that,
When adding comment using .comment()

`this.quoteModel.findOneAndUpdate(
      { _id: new mongoose.Types.ObjectId(quoteId) },
      newQuoteObj,
      {
        upsert: true,
        session,
        $comment: ctx.requestId,
      },
    ).comment("sadas");`
    
    But I received an error of:
    `comment cannot be used with findOneAndUpdate`
    
    Why is this? As I've seen in mongo docs, comment is allowed inside findAndModify function:
    https://www.mongodb.com/docs/manual/reference/command/findAndModify/

**Comment by vkarpov15** [comment]
This was fixed in #11793
